### ISSUE TITLE
Update CBOR benchmarks results and tag v2.5.0

### ISSUE THREAD

**Issue by x448** [fxamacker/cbor#390]
There are important new features, fixes, and performance improvements since v2.4.0.

@fxamacker What's the ballpark timeframe for tagging v2.5.0-rc or v2.5.0-beta2?

### Comparison of tip (on Feb 20, 2023) vs v2.4.0

I don't use Encoder/Decoder because your other API is faster, but this looks amazing for people who need it.

Amazing improvements in `NewEncoder().Encode()` but `Decode()` speed is a bit slower.  I didn't see equivalent benchmarks for `NewDecoder().Decode()`, so here's what I have so far.

```
$ benchstat benchNewEncoderEncode-v2.4.0.log benchNewEncoderEncode-f9e6291.log 
goos: linux
goarch: amd64
pkg: github.com/fxamacker/cbor/v2
cpu: 12th Gen Intel(R) Core(TM) i7-12700H
                                                     │ benchNewEncoderEncode-v2.4.0.log │  benchNewEncoderEncode-f9e6291.log  │
                                                     │              sec/op              │   sec/op     vs base                │
NewEncoderEncode/Go_bool_to_CBOR_bool-20                                   236.70n ± 2%   58.04n ± 1%  -75.48% (p=0.000 n=10)
NewEncoderEncode/Go_uint64_to_CBOR_positive_int-20                         238.00n ± 2%   63.93n ± 1%  -73.14% (p=0.000 n=10)
NewEncoderEncode/Go_int64_to_CBOR_negative_int-20                          238.65n ± 2%   64.88n ± 1%  -72.81% (p=0.000 n=10)
NewEncoderEncode/Go_float64_to_CBOR_float-20                               242.00n ± 2%   63.00n ± 1%  -73.97% (p=0.000 n=10)
NewEncoderEncode/Go_[]uint8_to_CBOR_bytes-20                               245.60n ± 1%   68.55n ± 1%  -72.09% (p=0.000 n=10)
NewEncoderEncode/Go_string_to_CBOR_text-20                                 243.20n ± 3%   68.39n ± 1%  -71.88% (p=0.000 n=10)
NewEncoderEncode/Go_[]int_to_CBOR_array-20                                  563.0n ± 2%   378.3n ± 0%  -32.81% (p=0.000 n=10)
NewEncoderEncode/Go_map[string]string_to_CBOR_map-20                        2.043µ ± 2%   1.906µ ± 2%   -6.75% (p=0.000 n=10)
geomean                                                                     349.7n        122.7n       -64.92%

                                                     │ benchNewEncoderEncode-v2.4.0.log │    benchNewEncoderEncode-f9e6291.log    │
                                                     │               B/op               │    B/op     vs base                     │
NewEncoderEncode/Go_bool_to_CBOR_bool-20                                     128.0 ± 0%     0.0 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_uint64_to_CBOR_positive_int-20                           128.0 ± 0%     0.0 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_int64_to_CBOR_negative_int-20                            128.0 ± 0%     0.0 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_float64_to_CBOR_float-20                                 128.0 ± 0%     0.0 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_[]uint8_to_CBOR_bytes-20                                 128.0 ± 0%     0.0 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_string_to_CBOR_text-20                                   128.0 ± 0%     0.0 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_[]int_to_CBOR_array-20                                   128.0 ± 0%     0.0 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_map[string]string_to_CBOR_map-20                         544.0 ± 0%   416.0 ± 0%   -23.53% (p=0.000 n=10)
geomean                                                                      153.4                    ?                       ¹ ²
¹ summaries must be >0 to compute geomean
² ratios must be >0 to compute geomean

                                                     │ benchNewEncoderEncode-v2.4.0.log │    benchNewEncoderEncode-f9e6291.log    │
                                                     │            allocs/op             │ allocs/op   vs base                     │
NewEncoderEncode/Go_bool_to_CBOR_bool-20                                     2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_uint64_to_CBOR_positive_int-20                           2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_int64_to_CBOR_negative_int-20                            2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_float64_to_CBOR_float-20                                 2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_[]uint8_to_CBOR_bytes-20                                 2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_string_to_CBOR_text-20                                   2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_[]int_to_CBOR_array-20                                   2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.000 n=10)
NewEncoderEncode/Go_map[string]string_to_CBOR_map-20                         28.00 ± 0%   26.00 ± 0%    -7.14% (p=0.000 n=10)
geomean                                                                      2.782                    ?                       ¹ ²
¹ summaries must be >0 to compute geomean
² ratios must be >0 to compute geomean
```

**Comment by fxamacker** [comment]
@x448 thanks for benchmarks.  Tagged [v2.5.0-beta2](https://github.com/fxamacker/cbor/releases/tag/v2.5.0-beta2) tonight.  It has fewer changes than [v2.5.0-beta](https://github.com/fxamacker/cbor/releases/tag/v2.5.0-beta).

I want to release v2.5.0 with one more PR but didn't have enough time to finish reviewing that PR this holiday weekend.  I can finish reviewing that PR next weekend.

**Comment by fxamacker** [comment]
Quick update: v2.5.0-beta5 passed several billions of execs fuzzing and I'd like to tag v2.5.0 release.  However, the README has a lot of outdated content such as codec comparisons that need to be removed or updated because they are old versions.

I'll push an update to the README today but I'm not sure if I'll have time to run benchmarks for inclusion in the README.  I think maybe I'll remove the benchmarks from the README and instead just point to an external document outside this repo so the README doesn't become a blocker for tagging new releases of this codec.

Thoughts?

**Comment by x448** [comment]
Good idea.  We can delete or comment out the parts of README that are outdated so we can have v2.5.0 sooner rather than later.

**Comment by fxamacker** [comment]
Released v2.5.0.  Closing this issue.
